packages/compass-import-export/src/components/import-preview.tsx

```diff
@@ -47,49 +59,101 @@ const cellUncheckedStyles = css({
-  setFieldType,
-  loaded,
+const rowIndexStyles = css({
+  minWidth: 0,
+  color: palette.gray.base,
+});
+
+const fieldTypeContainerStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+  gap: spacing[1],
+});
+
+const infoIconCSS = css({
+  height: spacing[3],
+  color: palette.gray.base,
+});
+
+type Field = {
+  path: string;
+  type: CSVParsableFieldType | 'placeholder';
+  checked: boolean;
+  result?: CSVField;
+};
+
+function MixedWarning({
+  result,
+  selectedType,
 }: {
-  fields: {
-    path: string;
-    type: string;
-    checked: boolean;
-  }[];
-  values: string[][];
-  onFieldCheckedChanged: (fieldPath: string, checked: boolean) => void;
-  setFieldType: (fieldPath: string, fieldType: string) => void;
-  loaded: boolean;
+  result: CSVField;
+  selectedType: CSVParsableFieldType;
 }) {
-  if (!loaded) {
-    debug('Preview unavailable: not loaded yet');
+  if (!['mixed', 'number'].includes(result.detected)) {
     return null;
   }
 
-  if (!Array.isArray(fields) || !Array.isArray(values)) {
-    debug('Preview unavailable: Fields or values is not an array', {
-      fields,
-      values,
-    });
+  // once the user manually changed the type, make the warning go away
+  if (result.detected !== selectedType) {
     return null;
   }
 
   return (
-    <>
-      <Body weight="medium">Specify Fields and Types</Body>
-      <Table
-        data={values}
-        columns={fields.map((field) => (
-          <TableHeader
-            key={`col-${field.path}`}
-            label={
-              <div
-                className={columnHeaderStyles}
-                data-testid={`preview-field-header-${field.path}`}
-              >
+    <Tooltip
+      align="top"
+      justify="middle"
+      delay={500}
+      trigger={({ children, ...props }) => (
+        <div {...props}>
+          {children}
+          <div className={infoIconCSS}>
+            <Icon glyph="InfoWithCircle"></Icon>
+          </div>
+        </div>
+      )}
+    >
+      <>
+        <p>
+          This field has{' '}
+          {selectedType === 'number'
+            ? 'mixed numeric types'
+            : 'mixed data types'}
+          :
+        </p>
+        <ul>
+          {Object.entries(result.types).map(([type, info]) => {
+            return (
+              <li key={type}>
+                {type} * {info.count}
+              </li>
+            );
+          })}
+        </ul>
+        <p>To standardise your data, select a different type.</p>
+      </>
+    </Tooltip>
+  );
+}
+
+function FieldHeader(
+  field: Field,
+  analyzed: boolean,
+  onFieldCheckedChanged: (fieldPath: string, checked: boolean) => void,
+  setFieldType: (fieldPath: string, fieldType: string) => void
+) {
+  return (
+    <TableHeader
+      key={`col-${field.path}`}
+      label={
+        <div
+          className={columnHeaderStyles}
+          data-testid={`preview-field-header-${field.path}`}
+        >
+          {field.type === 'placeholder' ? (
```

Comment:
Total small thing, feel free to ignore, might be a touch less code to write this instead as
```
{field.type !== 'placeholder' && (
  <>
  ...
```

Reply:
Aah yes that's left over from when this was a lot more complicated.